vmware/cloud-provider-for-cloud-director
# [VCDA-3729] Remove VIP from "virtualIPs" section during upgrade

* CAPVCD will preserve the "virtualIPs" section when upgrading the RDE from capvcdCluster:1.0.0 to capvcdCluster:1.1.0
* CPI should remove the specific IP from "entity.status.virtualIPs"  and populate them to "entity.status.cpi.vcdResourceSet"
* The previous code used to add IP to `vcdResourceSet` of CPI status
* This PR removes the specific IP which was moved to the new section from "entity.status.virtualIPs" section.

Testing done -
* Execute tests
* Create 1.1.1 CPI with CAPVCD 0.5.x, Upgrade CAPVCD, Upgrade CPI (recreate config-maps and deployment)
    * CPI should move virtual IPs one-by-one from "entity.status.virtualIPs" to "entity.status.cpi.vcdResourceSet

Signed-off-by: Aniruddha Shamasundar <aniruddha.9794@gmail.com>


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/cloud-provider-for-cloud-director/88)
